specification/hybridconnectivity/resource-manager/Microsoft.HybridConnectivity/stable/2023-03-15/hybridconnectivity.json

```diff
@@ -0,0 +1,1093 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EndpointResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Endpoint resource.",
+            "schema": {
+              "$ref": "#/definitions/EndpointResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Endpoints"
+        ],
+        "description": "Deletes the endpoint access to the target resource.",
+        "operationId": "Endpoints_Delete",
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "HybridConnectivityEndpointsDeleteDefault": {
+            "$ref": "./examples/EndpointsDeleteDefault.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "#/parameters/EndpointNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The endpoint is deleted successfully."
+          },
+          "204": {
+            "description": "The endpoint has been deleted already."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.HybridConnectivity/endpoints/{endpointName}/serviceconfigurations/{serviceConfigurationName}": {
+      "get": {
+        "tags": [
+          "Serviceconfigurations"
+        ],
+        "description": "Gets the details about the service to the resource.",
+        "operationId": "Serviceconfigurations_Get",
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "HybridConnectivityEndpointsServiceconfigurationsGetSSH": {
+            "$ref": "./examples/ServiceconfigurationsGetSSH.json"
+          },
+          "HybridConnectivityEndpointsServiceconfigurationsGetWAC": {
+            "$ref": "./examples/ServiceconfigurationsGetWAC.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
```

Comment:
There's a similar-in-spirit parameter defined in common-types for extension resources which is

 {
            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ScopeParameter"
          },

I don't know if you can actually use it.

Reply:
(Looks like you are already committed to the resourceUri approach, probably easier not to change that now!)

Reply:
That's right, the entire implementation has been done with resourceUri now.